src/core/geonode/qgsgeonoderequest.h

```diff
@@ -0,0 +1,112 @@
+/***************************************************************************
+    qgsgeonoderequest.h
+    ---------------------
+    begin                : Jul 2017
+    copyright            : (C) 2017 by Muhammad Yarjuna Rohmat, Ismail Sunni
+    email                : rohmat at kartoza dot com, ismail at kartoza dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSGEONODEREQUEST_H
+#define QGSGEONODEREQUEST_H
+
+#include "qgis.h"
+#include "qgis_core.h"
+#include <qnetworkreply.h>
+
+
+#include <QObject>
+#include <QUuid>
+
+struct CORE_EXPORT QgsServiceLayerDetail
+{
+#ifdef SIP_RUN
+  % TypeHeaderCode
+#include <qgsgeonoderequest.h>
+  % End
+#endif
+  QUuid uuid;
+  QString name;
+  QString typeName;
+  QString title;
+  QString wmsURL;
+  QString wfsURL;
+  QString xyzURL;
+};
+
+class CORE_EXPORT QgsGeoNodeRequest : public QObject
```

Comment:
This whole class should be properly doxymented (on class and method level) with added in etc..